pkg/restore/restore.go

```diff
@@ -101,6 +103,7 @@ type Restorer interface {
 // kubernetesRestorer implements Restorer for restoring into a Kubernetes cluster.
 type kubernetesRestorer struct {
 	restoreClient              velerov1client.RestoresGetter
+	client                     kbclient.Client
 	discoveryHelper            discovery.Helper
```

Comment:
let's delete discoveryHelper and dynamicFactory here so we are sure they are not used anywhere